### ISSUE TITLE
[whitespace/braces] false positive on tuple deconstruction

### ISSUE THREAD

**Issue by XDracam** [cpplint/cpplint#113]
When I write the following code:

```c++
for (auto [foo, bar] : baz)
```

I get the warning `Extra space before [  [whitespace/braces] [5]` But this feels wrong:

```c++
for (auto[foo, bar] : baz)
```

Is this a false positive from analyzing `operator[]`s?

**Comment by tkruse** [comment]
Yes, I believe the case you describe was not considered in the code https://github.com/cpplint/cpplint/blob/4f2119f3402b91c9a97dc247191129b14d246b8a/cpplint.py#L3512

> ```
>  # You shouldn't have spaces before your brackets, except maybe after
>  # 'delete []' or 'return []() {};'
> if Search(r'\w\s+\[', line) and not Search(r'(?:delete|return)\s+\[', line):
>    error(filename, linenum, 'whitespace/braces', 5,
>         'Extra space before [')
> ```

I believe cpplint generally does not handle the "auto" keyword, are there other cases that should be added?

**Comment by tkruse** [comment]
I believe this was recently fixed upstream in https://github.com/google/styleguide/pull/487
I will include this soon as https://github.com/cpplint/cpplint/pull/126
